mwdb/model/file.py

```diff
@@ -81,30 +91,81 @@ def get_or_create(cls, file, parent=None, metakeys=None, share_with=None):
                     app_config.mwdb.s3_storage_secret_key,
                     app_config.mwdb.s3_storage_region_name,
                     app_config.mwdb.s3_storage_secure,
-                ).put_object(app_config.mwdb.s3_storage_bucket_name, file_obj._calculate_path(), file, file_size)
+                ).put_object(
+                    app_config.mwdb.s3_storage_bucket_name,
+                    file_obj._calculate_path(),
+                    file.stream,
+                    file_size
+                )
             else:
                 file.save(file_obj._calculate_path())
 
+        file_obj.upload_stream = file.stream
         return file_obj, is_new
 
     def _calculate_path(self):
-        # upload_path must not be None
-        upload_path = app_config.mwdb.uploads_folder or ""
+        if app_config.mwdb.storage_provider == StorageProviderType.DISK:
+            upload_path = app_config.mwdb.uploads_folder
+        else:
+            upload_path = ""
+
         sample_sha256 = self.sha256.lower()
         
         if app_config.mwdb.hash_pathing:
             # example: uploads/9/f/8/6/9f86d081884c7d659a2feaa0c55ad015a3bf4f1b2b0b822cd15d6c15b0f00a08
             upload_path = os.path.join(upload_path, *list(sample_sha256)[0:4])
-    
+
         if app_config.mwdb.storage_provider == StorageProviderType.DISK:
             upload_path = os.path.abspath(upload_path)
             os.makedirs(upload_path, mode=0o755, exist_ok=True)
         return os.path.join(upload_path, sample_sha256)
 
+    def get_path(self):
+        """
+        Legacy method used to retrieve the path to the file contents.
+
+        Creates NamedTemporaryFile if mwdb-core uses different type of
+        storage than DISK and file size is too small to be written to
+        disk by Werkzeug.
+
+        Deprecated, use File.open() to get the stream with contents.
+        """
+        if app_config.mwdb.storage_provider == StorageProviderType.DISK:
+            # Just return path of file stored in local file-system
+            return self._calculate_path()
+
+        if not self.upload_stream:
+            raise ValueError("Can't retrieve local path for this file")
+
+        if type(self.upload_stream.name) is str:
```

Comment:
Humm.. I see that name can be provided in `bytes` as well (https://docs.python.org/3/library/io.html#io.FileIO).

But still I don't expect any inheritance here. Isn't it enough to check exact type in that case instead of `isinstance`? 🤔

Reply:
`isinstance` to me seems like a more "familiar" pattern. though I can see your point :+1: